### DISCUSSION TITLE
AttributeError: 'SentenceTransformer' object has no attribute 'embed_documents'

### DISCUSSION THREAD

**Discussion by ritish20** [PromtEngineer/localGPT#338] [Q&A] [answered]
Am getting below error while trying to run ingest.py.  note that i have downloaded the model and loaded it using below code 
from sentence_transformers import SentenceTransformer
    embeddings = SentenceTransformer('path to model')

i get the below error while trying to execute the ingest.py

from sentence_transformers import SentenceTransformer
embeddings = SentenceTransformer('hkunlp')
C:\Users\pradrit\dev\dbconda-env-pradrit\chatbot\lib\site-packages\bitsandbytes\cextension.py:34: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU quantization are unavailable.
  warn("The installed version of bitsandbytes was compiled without GPU support. "
'NoneType' object has no attribute 'cadam32bit_grad_fp32'

runfile('C:/Users//Ritish/DataScience Project Folder/chatbot/llama_chatbot/bloke/localGPT/ingest.py', wdir='C:/Users//Ritish/DataScience Project Folder/chatbot/llama_chatbot/bloke/localGPT')
2023-08-04 14:11:36,982 - INFO - ingest.py:121 - Loading documents from C:\Users\\Ritish\DataScience Project Folder\chatbot\llama_chatbot\bloke\localGPT/SOURCE_DOCUMENTS
2023-08-04 14:12:06,085 - INFO - ingest.py:130 - Loaded 1 documents from C:\Users\\Ritish\DataScience Project Folder\chatbot\llama_chatbot\bloke\localGPT/SOURCE_DOCUMENTS
2023-08-04 14:12:06,087 - INFO - ingest.py:131 - Split into 72 chunks of text
2023-08-04 14:12:06,089 - INFO - SentenceTransformer.py:66 - Load pretrained SentenceTransformer: hkunlp
2023-08-04 14:12:17,343 - INFO - SentenceTransformer.py:105 - Use pytorch device: cpu
2023-08-04 14:12:17,409 - INFO - __init__.py:88 - Running Chroma using direct local API.
2023-08-04 14:12:17,578 - WARNING - __init__.py:43 - Using embedded DuckDB with persistence: data will be stored in: C:\Users\\Ritish\DataScience Project Folder\chatbot\llama_chatbot\bloke\localGPT/DB
2023-08-04 14:12:17,751 - INFO - ctypes.py:22 - Successfully imported ClickHouse Connect C data optimizations
2023-08-04 14:12:17,874 - INFO - json_impl.py:45 - Using ujson library for writing JSON byte strings
2023-08-04 14:12:19,420 - INFO - duckdb.py:460 - loaded in 0 embeddings
2023-08-04 14:12:19,485 - INFO - duckdb.py:472 - loaded in 0 collections
Traceback (most recent call last):

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\users\\ritish\datascience project folder\chatbot\llama_chatbot\bloke\localgpt\ingest.py:161 in <module>
    main()

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\click\core.py:1130 in __call__
    return self.main(*args, **kwargs)

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\click\core.py:1055 in main
    rv = self.invoke(ctx)

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\click\core.py:1404 in invoke
    return ctx.invoke(self.callback, **ctx.params)

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\click\core.py:760 in invoke
    return __callback(*args, **kwargs)

  File c:\users\\ritish\datascience project folder\chatbot\llama_chatbot\bloke\localgpt\ingest.py:147 in main
    db = Chroma.from_documents(

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\langchain\vectorstores\chroma.py:446 in from_documents
    return cls.from_texts(

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\langchain\vectorstores\chroma.py:407 in from_texts
    chroma_collection = cls(

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\langchain\vectorstores\chroma.py:95 in __init__
    embedding_function=self._embedding_function.embed_documents

  File ~\dev\dbconda-env-pradrit\chatbot\lib\site-packages\torch\nn\modules\module.py:1614 in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(

AttributeError: 'SentenceTransformer' object has no attribute 'embed_documents'

**Comment by ritish20** [comment, marked as answer]
fixed it by give the offline model path directly in 

loading offline instruction model-
	change below code in Lib\site-packages\langchain\embeddings\huggingface.py file
	
            self.client = INSTRUCTOR(
                'self.model_name, cache_folder=self.cache_folder', **self.model_kwargs
            )
			
	to 
	            self.client = INSTRUCTOR(
                'C:/Users/pradrit/OneDrive - Deutsche Bank AG/Ritish/DataScience Project Folder/chatbot/llama_chatbot/bloke/localGPT/hkunlp/', **self.model_kwargs
            )

**Comment by Abhi5h3k** [comment]
I encountered a similar problem while utilizing a sentence-transformers. For those who are attempting the same and would like to download the model once for subsequent use, here's a suggestion:

Consider providing the **_path to your model dir_** as the **_cache_folder_** argument insted of editing the library code as it will cause issues on update.

**Example**: `HuggingFaceEmbeddings(cache_folder=model_folder_path)`

By following this approach, the model will be downloaded to the specified directory during the initial attempt, and then you can conveniently use it from that location onwards.

**Reply by Albert337** [reply]
> I encountered a similar problem while utilizing a sentence-transformers. For those who are attempting the same and would like to download the model once for subsequent use, here's a suggestion:
> 
> Consider providing the **_path to your model dir_** as the **_cache_folder_** argument insted of editing the library code as it will cause issues on update.
> 
> **Example**: `HuggingFaceEmbeddings(cache_folder=model_folder_path)`
> 
> By following this approach, the model will be downloaded to the specified directory during the initial attempt, and then you can conveniently use it from that location onwards.

Thanks, but when i fixed my code as your instaructions
![image](https://github.com/PromtEngineer/localGPT/assets/33714840/1c08af6d-751f-4b7f-8708-be9a5f5d873e)
i still get the error,
![image](https://github.com/PromtEngineer/localGPT/assets/33714840/5c15db14-a636-4fc7-af76-7083f9fef20e)
is there any other suggestions?Thanks a lot!

**Reply by NowhereDuk** [reply]
I get this error too, do u solve it?

**Reply by eternal-bug** [reply]
you can check there: [Use Chromadb with Langchain and embedding from SentenceTransformer model](https://github.com/langchain-ai/langchain/discussions/7818#discussioncomment-9847784)
